### PR TITLE
chore(android): update for Android 12

### PR DESCRIPTION
**Pull request by jquick-axway**
**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-28478

**Android Changes:**
- Add immutable/mutable flag to pending intent. (Prevents exception on Android 12.)
- Add "course" location permission to manifest. (Required by Android 12 when requesting "fine" permission.)
- Fixed example code crashes on Android.
- Updated module version to 5.0.1.

**iOS Changes:**
- Updated module version to 4.0.0.
- Dropped 32-bit support.
- Changed min Titanium SDK version to 10.0.0.

### CODE REVIEW

File: android/manifest [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # this is your module manifest and used by Titanium
 # during compilation, packaging, distribution, etc.
 #
-version: 4.0.0
+version: 5.0.1
```
Comment (garymathews) [inline android/manifest:5]: I think this can be `5.0.0` since the `package.json` is indepdendent of this

Comment (jquick-axway) [reply android/manifest]: Done

**Review by garymathews — APPROVED** [review]
CR: **PASS**

**Comment by lokeshchdhry** [comment]
FR Passed.

**Pull request merged**